graphbolt/CMakeLists.txt

```diff
@@ -76,11 +77,31 @@ include_directories(BEFORE ${BOLT_DIR}
                            "../third_party/pcg/include"
                            "../third_party/phmap")
 target_link_libraries(${LIB_GRAPHBOLT_NAME} "${TORCH_LIBRARIES}")
+
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
   if(USE_LIBURING)
-    target_include_directories(${LIB_GRAPHBOLT_NAME} PRIVATE "../third_party/liburing/src/include")
-    get_filename_component(PARENT_DIR "${CMAKE_SOURCE_DIR}" DIRECTORY)
-    target_link_libraries(${LIB_GRAPHBOLT_NAME} ${PARENT_DIR}/build/third_party/liburing/lib/liburing.a)
+    add_definitions(-DHAVE_LIBRARY_LIBURING)
+    include(ExternalProject)
+    set(LIBURING_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/third_party/liburing)
+    set(LIBURING_C_COMPILER "${CMAKE_C_COMPILER} -w")
+    ExternalProject_Add(
+      liburing
+      SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/liburing
+      CONFIGURE_COMMAND <SOURCE_DIR>/configure --cc=${LIBURING_C_COMPILER} --cxx=${CMAKE_CXX_COMPILER} --prefix=/
+      BUILD_COMMAND bash -c "make -j 4"
```

Comment:
why limit 4 processes for build?

Reply:
I haven't added this myself. Old code had it. It builds crazy fast so it does not really matter.

Reply:
When we use `make -j`, people with limited RAM can not build our project so it is better to limit it if the build is incredibly fast anyway.